tmail-backend/rate-limiter/rate-limiter-api/src/main/scala/com/linagora/tmail/rate/limiter/api/RateLimitationPlanRepository.scala

```diff
@@ -0,0 +1,63 @@
+package com.linagora.tmail.rate.limiter.api
+
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+import com.google.common.base.Preconditions
+
+import scala.collection.concurrent.Map
+
+trait RateLimitationPlanRepository {
+
+  def create(creationRequest: RateLimitingPlanCreateRequest): Publisher[RateLimitingPlan]
+
+  def update(resetRequest: RateLimitingPlanResetRequest): Publisher[RateLimitingPlan]
+
+  def get(id: RateLimitingPlanId): Publisher[RateLimitingPlan]
+
+  def isPlanExisted(id: RateLimitingPlanId): Publisher[Boolean]
+
+  def list(): Publisher[RateLimitingPlan]
+
+  def getTransition(id: RateLimitingPlanId, operationLimitationsName: String): Publisher[OperationLimitations]
```

Comment:
Is this necessary?

`repo.get(id)` retrieves the plan which I can easily find operations in.

I feel ike this do not delivers value

Reply:
I remember it. 
When grooming we explained it. 
It will help Cassandra implement queries faster.  (This method will be invoked much time when running)

```
PK: RATE_LIMITING_PLAN_ID
CK: OPERATION_NAME
```

Reply:
I removed it on this pr in the last commit, If it has value, I will revert it